include/zephyr/drivers/clock_control/nrf_clock_control.h

```diff
@@ -292,6 +292,30 @@ int nrf_clock_control_cancel_or_release(const struct device *dev,
 	return api->cancel_or_release(dev, spec, cli);
 }
 
+/** @brief Request the HFXO from Zero Latency Interrupt context.
+ *
+ * Function is optimized for use in Zero Latency Interrupt context.
+ * It does not give notification when the HFXO is ready, so each
+ * user must put the request early enough to make sure the HFOX
+ * ramp-up has finished on time.
+ *
+ * Function does reference counting so the caller must ensure that
+ * every nrf_clock_control_request_zli has matching nrf_clock_-
+ * control_hfxo_release call.
+ */
+void nrf_clock_control_hfxo_request(void);
+
+/** @brief Release the HFXO from Zero Latency Interrupt context.
+ *
+ * Function is optimized for use in Zero Latency Interrupt context.
+ * It releases a HFXO provided by pointer to device.
+ *
+ * The function use must be preceded by nrf_clock_control_hfxo_request
+ * call, because it uses basic reference counting to make sure the
+ * HFXO is relased when there is no more pending requests.
```

Comment:
```suggestion
 * Calls to this function must be coupled with prior calls
 * to nrf_clock_control_hfxo_request(), because it uses basic
 * reference counting to make sure the HFXO is released when
 * there are no more pending requests.
```